src/dbos-executor.ts

```diff
@@ -514,16 +582,53 @@ export class DBOSExecutor {
       this.logger.error(`Failed to find inputs for workflowUUID: ${workflowUUID}`);
       throw new DBOSError(`Failed to find inputs for workflow UUID: ${workflowUUID}`);
     }
-    const wfInfo: WorkflowInfo<any, any> | undefined = this.workflowInfoMap.get(wfStatus.workflowName);
+    const parentCtx = this.#getRecoveryContext(workflowUUID, wfStatus);
+
+    const wfInfo: WorkflowInfo | undefined = this.workflowInfoMap.get(wfStatus.workflowName);
 
-    // If wfInfo is undefined, then it means it's a temporary workflow.
-    // TODO: we need to find the name of that function and run it.
-    if (!wfInfo) {
-      throw new DBOSError(`Cannot find workflow info for UUID ${workflowUUID}`);
+    if (wfInfo) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+      return this.workflow(wfInfo.workflow, { workflowUUID: workflowUUID, parentCtx: parentCtx ?? undefined }, ...inputs);
+    }
+
+    // Should be temporary workflows. Parse the name of the workflow.
+    const wfName = wfStatus.workflowName;
+    const nameArr = wfName.split("-");
+    if (!nameArr[0].startsWith(DBOSExecutor.tempWorkflowName)) {
+      throw new DBOSError(`This should never happen! Cannot find workflow info for UUID ${workflowUUID}, name ${wfName}`);
```

Comment:
This message could be made more explicit by saying what should never happen (finding a non-temporary workflow)